windowPreview.js

```diff
@@ -388,26 +392,35 @@ class WindowPreviewMenuItem extends PopupMenu.PopupBaseMenuItem {
         return [width, height, scale];
     }
 
-    _cloneTexture(metaWin){
-
-        let mutterWindow = metaWin.get_compositor_private();
+    _updateWindowPreviewSize() {
+        // This gets the actual windows size for the preview
+        [this._width, this._height, this._scale] = this._getWindowPreviewSize();
```

Comment:
Why not move this logic into the update handler? This utility doesn't appear to be used elsewhere.

Reply:
I preferred to change too much unneeded code here, at the end having an extra function should not harm, while makes it clearer to read the update routine.